packages/editor-ui/src/components/ContactPromptModal.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+	<Modal
+		:name="modalName"
+		:eventBus="modalBus"
+		:center="true"
+		:closeOnPressEscape="false"
+		:beforeClose="closeDialog"
+		customClass="contact-prompt-modal"
+		width="460px"
+	>
+		<template slot="header">
+			<n8n-heading tag="h2" size="xlarge" color="text-dark">{{ title }}</n8n-heading>
+		</template>
+		<template v-slot:content>
+			<div :class="$style.description">
+				<n8n-text size="medium" color="text-base">{{ description }}</n8n-text>
+			</div>
+			<n8n-input
+				v-model="email"
+				placeholder="Your email address"
+			/>
+			<div :class="$style.disclaimer">
+				<n8n-text size="small" color="text-base">David from our product team will get in touch personally.</n8n-text>
+			</div>
+		</template>
+		<template v-slot:footer>
+			<div :class="$style.footer">
+				<n8n-button label="Send" float="right" @click="send" :disabled="!isEmailValid" />
+			</div>
+		</template>
+	</Modal>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import mixins from "vue-typed-mixins";
+import { mapGetters } from 'vuex';
+
+import { VALID_EMAIL_REGEX } from '@/constants';
+import { workflowHelpers } from "@/components/mixins/workflowHelpers";
+import Modal from "./Modal.vue";
+
+export default mixins(workflowHelpers).extend({
+	components: { Modal },
+	name: "ContactPromptModal",
+	props: ["modalName"],
+	data() {
+		return {
+			email: '',
+			modalBus: new Vue(),
+		};
+	},
+	computed: {
+		...mapGetters({
+			promptData: 'settings/getPromptsData',
+		}),
+		title() : string {
+			if (this.promptData && this.promptData.title) {
+				return this.promptData.title;
+			} else {
+				return 'You’re a power user 💪';
+			}
+		},
+		description() : string {
+			if (this.promptData && this.promptData.message) {
+				return this.promptData.message;
+			} else {
+				return 'Your experience with n8n can help us improve - for you and our entire community.';
+			}
+		},
+		isEmailValid(): boolean {
+			return VALID_EMAIL_REGEX.test(String(this.email).toLowerCase());
+		},
+	},
+	methods: {
+		closeDialog(): void {
+			if (!this.isEmailValid) {
+				this.$telemetry.track('User closed email modal', { instance_id: this.$store.getters.instanceId, email: null });
+			}
+			this.$store.commit('ui/closeTopModal');
+		},
+		send(): void {
+			if (this.isEmailValid) {
+				this.$store.dispatch('settings/submitContactInfo', this.email);
```

Comment:
disptach returns a promise. we should add `await` here for response from service and add a loading state to button

Reply:
we should also decide how to handle errors, in this case it just shows thank you